spec/views/users_index_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'Users Index', type: :feature do
+  root_path = '/users'
+
+  scenario 'Shows the name of every user' do
+    visit root_path
+    expect(page).to have_content('Tom')
+    expect(page).to have_content('Lilly')
+  end
+
+  scenario 'shows two user images' do
+    visit root_path
+    expect(page).to have_selector('img', count: 2)
+    expect(page).to have_selector('img[src="https://randomuser.me/api/portraits/men/1.jpg"]')
+  end
+
+  scenario 'shows correct number of posts for each user' do
+    visit root_path
+    expect(page).to have_selector('span', text: '4')
+    expect(page).to have_selector('span', text: '0')
+  end
+
+  scenario 'clicks on a use redirect to user profile' do
+    visit root_path
+    click_link(href: '/users/1')
+    expect(page).to have_current_path('/users/1')
+  end
+end
```

Comment:
- [ ] When I run the tests, I get the above response in my terminal:

![image](https://user-images.githubusercontent.com/77106962/231592368-fedd3e8f-0eef-439f-9661-928c65111a17.png)

#### Even though the tests pass, I am running them in the **_development environment_** with the data you provided in the `db/seed.rb` file. Ideally, these tests should be run in the **_test environment_** with test data provided for the examples in each spec file. When I run these tests in the **_test environment_**, I get these responses:

![image](https://user-images.githubusercontent.com/77106962/231593244-c9c711e7-ce33-4c12-8e75-3cdad5d0008f.png)

#### The tests fail in the **_test environment_** because, in that environment, no data for the `User` or `Post` models were provided. I would suggest that you create data in the spec files while writing the tests. This will make the tests pass.

#### Hint: You can use the `before :each` method before writing your assertion tests. 👍 😎 

```ruby
 before :each do
    @user1 = User.create(name: 'Jade', email: 'jade@example.com', bio: 'Test bio')

    Post.create(id: 4, title: 'post 1', text: 'text 1', likes_counters: 0, comments_counter: 0, user_id: @user1.id)
    Post.create(id: 5, title: 'post 2', text: 'text 2', likes_counters: 0, comments_counter: 0, user_id: @user1.id)
    Post.create(id: 6, title: 'post 3', text: 'text 3', likes_counters: 0, comments_counter: 0, user_id: @user1.id)
  end
```